include/deal.II/lac/utilities.h

```diff
@@ -0,0 +1,160 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__lac_utilities_h
+#define dealii__lac_utilities_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/lac/lapack_templates.h>
+#include <deal.II/lac/lapack_support.h>
+#include <deal.II/lac/vector_memory.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Utilities
+{
+  /**
+   * A collection of linear-algerbra utilities
+   */
+  namespace Lac
+  {
+
+    /**
+     * Estimate an upper bound of the spectrum of @p op by a @p k -step
+     * Lanczos process starting from the initial vector @p v0. Typical
+     * values of @p k are below 10.
+     *
+     * @p vector_memory is used to allocate memory for temporary vectors.
+     *
+     * @note this function uses Lapack routines to compute the largest
+     * eigenvalue of the tridiagonal Lanczos matrix T. The returned
+     * value is $ ||T||_2 + ||f||_2$.
+     *
+     * @author Denis Davydov, 2017
+     */
+    template <typename OperatorType, typename VectorType>
+    double Lanczos_largest_eigenvalue(const OperatorType &op,
```

Comment:
Can you state in the documentation what kind of interface `OperatorType` needs to provide?